src/plugins/intel_gna/src/common/tools.cpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "tools.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace common {
+
+bool fp32eq(float p1, float p2, float accuracy) {
```

Comment:
Why don't do it using template?

Reply:
It also will allow us to avoid static_cast

Reply:
currently cast is not required there can be warning regarding precision.

<typename T>
bool fp32eq(T p1, float p2, T accuracy) {
if we use template there will be ambiguity when  you will use  p1 as float and p2 as double so you will need to cast.

for 
<typename T, typename U>
bool fp32eq(T p1, float p2, U accuracy) {
There will be chosen precision for one of the types  T or U(in case they differ) and it maybe it will be not what we would expect.

Reply:
@marcinkusm I don't think that we should cover comparison of variables with different types. So it should be something like
```
<typename T>
bool is_fp_equal(T p1, T p2, T accuracy) {
```

Reply:
done